Add RotatingQuotes component to the landing page

avikalpg opened the pull request:
Quotes generate confidence in the users. It is a form of social validation. 

UX idea:
1. The landing hero tells the main value proposition and gives a high-level overview of the features
2. Read quotes from some real people to better understand the value proposition - social validation
3. See the features in more detail
4. See the list of companies that already use/trust Vibinex - social validation

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Added a rotating quotes section with smooth animated transitions and automatic rotation.
  - Quotes can show author, title, and optional subtitle when provided.
  - Updated home page layout so the rotating quotes appear prominently before the customer logos.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

avikalpg committed:
ui: landing: add RotatingQuotes component to enhance landing page engagement

avikalpg commented on pages/index.tsx:
```diff
@@ -13,6 +13,27 @@ import { getAuthUserId, getAuthUserName, getAuthUserEmail } from '../utils/auth'
 import JoinDiscord from '../views/JoinDiscord'
+import RotatingQuotes from '../views/RotatingQuotes'
+
+const quotes = [
+	{
+		text: "You are preparing the antidote to vibe coding",
+		author: "Vikas Kumar",
+		title: "Founder, DataviCloud",
+	},
+	{
+		text: "AI adoption for coding boosts perceived productivity but reduces delivery throughput",
+		author: "DORA Research @ Google",
+		title: "October 2024: Accelerate State of DevOps"
+	},
+	{
+		text: "AI agents will bullshit you if you can't call them out",
+		author: "Jared Freidman",
+		title: "Managing Partner, Y Combinator",
+		subtitle: "In the video titled 'Vibe Coding Is The Future'"
+	}
+	// Add more quotes here as needed
+];
```
There is also a DORA report launched in 2024 with the same title. The insight mentioned in the quote is from the 2024 report and does not exist in the 2023 report.

avikalpg committed:
ui: fix: RotatingQuotes - handled the case when the quotes array is empty

tapishr commented:
Getting this - 
![image](https://github.com/user-attachments/assets/95d4e71b-cd26-4a95-91d8-86209b0e61d2)

tapishr reviewed (changes requested):
Doesn't look correct in dark mode

Commit:
fix: use semantic color tokens in RotatingQuotes for dark mode compatibility

Replaced hardcoded Tailwind gray classes (text-gray-900, text-gray-700, etc.)
with semantic CSS-variable-backed tokens (text-foreground, text-muted-foreground)
that automatically adapt to light/dark mode via globals.css.

Also replaced straight quote characters with proper HTML entities (ldquo/rdquo)
for typographic correctness.

Addresses dark mode feedback from Tapish's review.

Commit:
fix: return null for empty quotes array instead of rendering empty container
